editor/project_manager.cpp

```diff
@@ -904,9 +904,23 @@ class ProjectDialog : public ConfirmationDialog {
 			default_renderer_type = EditorSettings::get_singleton()->get_setting("project_manager/default_renderer");
 		}
 
+#if defined(WEB_ENABLED)
+		// Web platform only supports `gl_compatibility`. Ignore the above editor setting's value.
+		default_renderer_type = "gl_compatibility";
+#elif defined(ANDROID_ENABLED)
+		// Mobile platforms only support `mobile` and `gl_compatibility`.
+		// Ignore the above editor setting's value if set to an incompatible value.
+		if (default_renderer_type == "forward_plus") {
+			default_renderer_type = "mobile";
+		}
+#endif
+
 		Button *rs_button = memnew(CheckBox);
 		rs_button->set_button_group(renderer_button_group);
 		rs_button->set_text(TTR("Forward+"));
+#if defined(ANDROID_ENABLED) || defined(WEB_ENABLED)
```

Comment:
The Forward+ works on mobile, I don't think it should be disabled

Reply:
The description text is written as "Supports desktop platforms only". Do we have a reasonable guarantee that Forward+ will run on every mobile device that supports the Mobile rendering method? If not, I wouldn't allow selecting it.

Reply:
I strongly disagree with your reasoning here. 

There are bound to be some devices that trip on the Forward+ renderer but not the Mobile renderer. Just as there are bound to be devices that trip on the Mobile renderer but not the Forward+ renderer. In either case, it is up to the user to select the option that works for them. We shouldn't be taking options away from users